File: 2024-2/Proyecto DAO SoftPub .NET/SoftPub/SoftPubGrupoModel/Grupo.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftPubGrupoModel
{
    public class Grupo
    {
        private int? idGrupo;
        private string nombre;
        private string sigla;
        private BindingList<Linea> lineas;           

        public Grupo()
        {
            this.idGrupo = null;
            this.nombre = null;
            this.sigla = null;
            this.lineas = new BindingList<Linea>();
        }

        public Grupo(int? idGrupo, string nombre, string sigla)
        {
            this.idGrupo = idGrupo;
            this.nombre = nombre;
            this.sigla = sigla;
            this.lineas = new BindingList<Linea>();
        }

        public int? IdGrupo { get => idGrupo; set => idGrupo = value; }
        public string Nombre { get => nombre; set => nombre = value; }
        public string Sigla { get => sigla; set => sigla = value; }
        public BindingList<Linea> Lineas { get => lineas; set => lineas = value; }
    }
}

File: 2024-2/0684/unidad07.b - Programación distribuida con BD/SoftPub_Back-end/SoftPubGrupoModel/src/pe/edu/pucp/softpub/grupo/model/Grupo.java
package pe.edu.pucp.softpub.grupo.model;

import java.util.ArrayList;

/**
 *
 * Autor: andres
 */
public class Grupo {

    private Integer idGrupo;
    private String nombre;
    private String sigla;
    protected ArrayList<Linea> lineas;

    public Grupo(Integer idGrupo, String nombre, String sigla) {
        this.idGrupo = idGrupo;
        this.nombre = nombre;
        this.sigla = sigla;
        this.lineas = null;
    }

    public Grupo() {
        this.idGrupo = null;
        this.nombre = null;
        this.sigla = null;
        this.lineas = null;
    }

    /**
     * @return the idGrupo
     */
    public Integer getIdGrupo() {
        return idGrupo;
    }

    /**
     * @param idGrupo the idGrupo to set
     */
    public void setIdGrupo(Integer idGrupo) {
        this.idGrupo = idGrupo;
    }

    /**
     * @return the nombre
     */
    public String getNombre() {
        return nombre;
    }

    /**
     * @param nombre the nombre to set
     */
    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    /**
     * @return the sigla
     */
    public String getSigla() {
        return sigla;
    }

    /**
     * @param sigla the sigla to set
     */
    public void setSigla(String sigla) {
        this.sigla = sigla;
    }
}
